### PR TITLE
feat(memory): add snapshot diff command

### LINKED ISSUE #1852: feat(memory): add memory snapshot diff command
## Description
Implement an LLM-friendly, independently shippable backlog item for **add memory snapshot diff command**. This issue should stay narrow enough for one agent/PR while producing durable behavior, tests, and documentation where appropriate.

## Affected locations
- Memory storage, retrieval, injection, and redaction paths
- Project/profile scoped memory policy
- Memory audit and migration tooling

## Impact
Without this work, future agents and operators will keep rediscovering the same gap, creating avoidable coordination cost, reliability risk, or ambiguous handoffs.

## Recommendation
Implement the scoped behavior with deterministic tests, concise docs, and structured output that PM/liveness tooling can consume.

## Goals
Deliver add memory snapshot diff command as a discrete capability or hardening change.
Make the behavior deterministic enough for autonomous workers to verify.
Keep the implementation small enough for a one-issue worker and one PR.
Capture operator/user-facing behavior in docs or help text when applicable.

## Expected outcomes
add memory snapshot diff command is available, documented, and test-backed.
Future PM/worker handoffs can point at this issue without needing extra interpretation.
Failure modes are explicit and produce actionable messages instead of silent drift.
The change does not broaden into adjacent backlog work or unrelated refactors.

## Suggested implementation slices
Inventory the current code/docs paths that own this behavior.
Add or update the narrow model/config/schema needed for the feature or guardrail.
Implement the smallest runtime/docs change that satisfies the acceptance criteria.
Add focused fixtures/tests for success, failure, and edge cases.
Update docs/help output and handoff notes so later agents can use it.

## Acceptance criteria
- [ ] There is a dedicated implementation or documentation change for add memory snapshot diff command.
- [ ] At least one automated test, fixture, or deterministic verifier covers the intended behavior.
- [ ] At least one negative/edge case is covered so the failure mode is explicit.
- [ ] Docs/help text or inline operator guidance explains how to use or interpret the behavior.
- [ ] The PR description links this issue and summarizes verification evidence.
- [ ] No unrelated production behavior is changed outside the stated scope.

## LLM handoff notes
- Treat this as a one-issue/one-PR unit.
- Start by searching for existing implementations and tests before adding new surfaces.
- Prefer structured outputs and fixtures over prose-only behavior.
- If this overlaps an existing open issue, comment with evidence and stop for PM deduplication instead of creating an unscoped implementation.
- Do not perform broad refactors or change unrelated worker/PM behavior while implementing this ticket.

## Source
Created by the 2026-07-11 fourth strategic backlog expansion run requesting 100 issues. Topic: **Persistent memory for agents**.

### PR DESCRIPTION
**Pull request by djm204**
Closes #1852\n\n## Summary\n- add `frankenbeast memory snapshot-diff <before> <after>` CLI parsing and help text\n- validate both inputs as BrainSnapshot JSON and emit a deterministic structured JSON diff for working memory, episodic events, checkpoint, and metadata\n- add focused unit coverage for parsing, success output, and invalid snapshot failure modes\n\n## Test plan\n- `npm run test --workspace @franken/orchestrator -- tests/unit/cli/memory-snapshot-diff.test.ts`\n- `npm run typecheck --workspace @franken/orchestrator`\n\nNotes: fbeast MCP tools were not available in this worker, so I followed AGENTS.md with normal git/gh/terminal verification.

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/cli/run.ts [hunk 1]
```diff
@@ -1030,6 +1031,21 @@ export async function main(): Promise<void> {
     return;
   }
 
+  if (args.subcommand === 'memory') {
+    try {
+      await handleMemoryCommand({
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/cli/run.ts]: Fixed in 8df237dc by routing the `memory` subcommand before the banner/config/logger/scaffold path, so `memory snapshot-diff` writes only the structured JSON response. Added a main() regression test that parses stdout and asserts no banner/config log is emitted.

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/cli/run.ts [hunk 2]
```diff
@@ -932,6 +933,21 @@ export async function main(): Promise<void> {
     return;
   }
 
+  if (args.subcommand === 'memory') {
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/cli/run.ts]: Current-head follow-up fixed this before merge: commit 8df237dc routes the memory subcommand before the banner/config/logger/scaffold path, and CI plus the 2026-07-14T08:08:06Z Codex pass on head 8df237dca7037747b65dc71489722118bd369cba were clean. Resolving this stale pre-fix thread for audit hygiene.
